clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/VenicePushJob.java

```diff
@@ -2307,12 +2313,56 @@ void validateValueSchema(
         .info("Validating value schema: {} for store: {}", pushJobSchemaInfo.getValueSchemaString(), setting.storeName);
 
     SchemaResponse getValueSchemaIdResponse;
-
     if (setting.enableWriteCompute) {
-      getValueSchemaIdResponse = ControllerClient.retryableRequest(
-          controllerClient,
-          setting.controllerRetries,
-          c -> c.getValueOrDerivedSchemaId(setting.storeName, pushJobSchemaInfo.getValueSchemaString()));
+      if (!isUpdateSchema(pushJobSchemaInfo.getValueSchemaString())) {
+        MultiSchemaResponse multiSchemaResponse = ControllerClient.retryableRequest(
+            controllerClient,
+            setting.controllerRetries,
+            c -> c.getAllValueAndDerivedSchema(setting.storeName));
+        StoreResponse storeResponse = ControllerClient
+            .retryableRequest(controllerClient, setting.controllerRetries, c -> c.getStore(setting.storeName));
+
+        if (storeResponse.isError()) {
+          throw new VeniceException("Store does not exist: " + setting.storeName);
+        }
+        if (multiSchemaResponse.isError()) {
+          throw new VeniceException("Unable to retrieve schemas for store: " + setting.storeName);
+        }
+        int supersetSchemaId = storeResponse.getStore().getLatestSuperSetValueSchemaId();
```

Comment:
I know this, but seems like it is not set in current API code path... I am kind of hesitated about this -> If I changed the implementation that's fine, but this change will be depending on controller deployment (that's ......kind of OK but becomes slower to rollout) What do you think?

Reply:
I see..
Let us add some comments to mention this behavior since the current logic is a little confusing.
It is good to update this field in the existing endpoint in the future, and let us cut a ticket to track this.
Once this endpoint is ready, VPJ can migrate to this single endpoint.

Reply:
Sure, let me add some doc and create a ticket.

Reply:
Updated the java doc and add a ticket in WC epic as it is not very suitable for BB queue.